### DISCUSSION TITLE
pip_requirements with git reference without username not working

### DISCUSSION THREAD

**Discussion by mikhailadvani** [renovatebot/renovate#15230] [Request Help] [answered]
For https references, the username is something that we will not be hardcoding in requirements.txt since every developer and CI will need a different username. What would be the best way to operate in such scenarios? One idea is to modify the regex and use RENOVATE_HOST_RULES for configuring credentials. I have implemented a proof of concept here - https://github.com/renovatebot/renovate/pull/15231

**Comment by HonkingGoose** [comment, marked as answer]
Marking the PR as the answer to the discussion:

- https://github.com/renovatebot/renovate/pull/15231
